build-and-test.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+set -e 	# Exit immediately upon failure
+set -o pipefail  # Carry failures over pipes
+
+repo_root="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+docker_repo="microsoft/dotnet-nightly"
+
+pushd "${repo_root}" > /dev/null
+
+find . -path './.git' -prune -o -path './test' -prune -o -path './update-dependencies' -prune -o -name 'Dockerfile' -printf '%p\t%d\n' | egrep -v 'nanoserver|windowsservercore' | sort -n -k2 | while read line
```

Comment:
>-printf [](start = 116, length = 7)

Does this work on OSX?  Alternatively you could do something like the following which I think will work on OSX.  This approach avoids the need to track the directory depth.

find . -name Dockerfile -print0 | xargs -0 -n1 dirname | egrep -v 'nanoserver|windowsservercore' | sort | while read line
do
    dockerfile_dir=${line}
    tag="${docker_repo}:$( sed -e 's/\.\///' -e 's/debian\///' -e 's/debian/sdk/' -e 's/\//-/g' <<< "${dockerfile_dir}" )"
    echo "----- Building ${tag} -----"
    docker build -t "${tag}" "${dockerfile_dir}"
done